### PR TITLE
iTop HELM chart

### PR DESCRIPTION
**Pull request by finaldzn**
Hello,
First of all thank you for your itop docker image, we are using it and we thank you for it.

We manage all our services on a kubernetes cluster and manage the applications using helm charts.

In order for us to use your image cleanly, we created a itop helm chart.

Please let me know if you need something or have questions.

You can also upload this chart to a chart museum in order to have it easily available to users.

Thank you

### CODE REVIEW

File: helm-chart/.helmignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+*.swp
+*.bak
+*.tmp
+*.orig
+*~
+# Various IDEs
+.project
+.idea/
+*.tmproj
+.vscode/
```
Comment (Hipska) [inline helm-chart/.helmignore]: Are these relevant for this helm-chart? I don't think so. Please remove in that case.

File: helm-chart/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+volumeMounts:
+            - mountPath: "/var/www/html/data"
+              name: itop-data-pv
+            - mountPath: "/var/www/html/env-production"
+              name: itop-env-pv
+            - mountPath: "/var/www/html/log"
+              name: itop-log-pv
+            - mountPath: "/var/www/html/conf"
+              name: itop-conf-pv
+```
```
Comment (Hipska) [inline helm-chart/README.md]: Please fix indentation and markup language. (also counts for other instances in this file.)
```suggestion
```yaml
volumeMounts:
   - mountPath: "/var/www/html/data"
     name: itop-data-pv
   - mountPath: "/var/www/html/env-production"
     name: itop-env-pv
   - mountPath: "/var/www/html/log"
     name: itop-log-pv
   - mountPath: "/var/www/html/conf"
     name: itop-conf-pv
```

```

File: helm-chart/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+### Values.yaml
+
+You can edit the values.yaml file directly or use the set directive with the helm cli tool.
+
+```
+image:
+  repository: supervisions/itop
+  pullPolicy: Always
+  tag: 3.0.0
```
Comment (Hipska) [inline helm-chart/README.md]: I think I would prefer to use this tag, so the helm chart or docs doesn't need to be updated every release.
```suggestion
  tag: 3.0
```

File: helm-chart/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+## HELM CHART
+
+This is a helm chart used to easily deploy itop to a K8S cluster using [HELM](https://helm.sh/).
```
Comment (Hipska) [inline helm-chart/README.md]:
```suggestion
This is a helm chart used to easily deploy iTop to a K8S cluster using [HELM](https://helm.sh/).

```

File: helm-chart/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+## HELM CHART
```
Comment (Hipska) [inline helm-chart/README.md]:
```suggestion
## HELM Chart
```

File: helm-chart/README.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+environment:
+  db_host: <mysql-server-url> "mysql-chart.database.svc.cluster.local"
+  db_name: <mysql-db-name>
+  db_pwd: <mysql-db-pwd>
+  db_user: <mysql-db-user>
+```
+
+### Pod security
+
+In order to allow itop to access the itop, adding specific security context for the pod was needed, those can be found in the deployment.yaml file
```
Comment (Hipska) [inline helm-chart/README.md]: > itop to access the itop

What do you mean with that?

File: helm-chart/README.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+If you want to know more about the do and don't i invite you to read the k8s documentation [here](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/)
+
+
+### Adding your own config-itop.php
+
+You can create a configmap and add it to the pod deployment in order to have the config-itop.php file added before installing itop.
+
+This is not recommanded.
```
Comment (Hipska) [inline helm-chart/README.md]:
```suggestion
This is not recommended.
```

File: helm-chart/README.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+### What is not added
+
+A mysql chart link in order to also have a db deployed to go with the itop instance.
```
Comment (Hipska) [inline helm-chart/README.md]:
```suggestion
A mysql chart link in order to also have a db deployed to go with the iTop instance.
```

File: helm-chart/templates/NOTES.txt [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get --namespace {{ .Release.Namespace }} svc -w {{ include "itop-chart.fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ include "itop-chart.fullname" . }} --template "{{"{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"}}")
+  echo http://$SERVICE_IP:{{ .Values.service.port }}
+{{- else if contains "ClusterIP" .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app.kubernetes.io/name={{ include "itop-chart.name" . }},app.kubernetes.io/instance={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
+  export CONTAINER_PORT=$(kubectl get pod --namespace {{ .Release.Namespace }} $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+  echo "Visit http://127.0.0.1:8080 to use your application"
+  kubectl --namespace {{ .Release.Namespace }} port-forward $POD_NAME 8080:$CONTAINER_PORT
+{{- end }}
```
Comment (Hipska) [inline helm-chart/templates/NOTES.txt]: Is this file needed? And if so, please add som more comment or description what is done here..

File: helm-chart/templates/deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+              containerPort: 80
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
```
Comment (Hipska) [inline helm-chart/templates/deployment.yaml]: iTop has a `status.php` endpoint. Maybe something can be done with that?

https://github.com/Combodo/iTop/blob/develop/webservices/status.php

File: helm-chart/values.yaml [hunk 11]
```diff
@@ -0,0 +1,92 @@
+# Default values for itop-chart.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: supervisions/itop
+  pullPolicy: Always
+  tag: 3.0.0
```
Comment (Hipska) [inline helm-chart/values.yaml]:
```suggestion
  tag: 3.0
```

**Review by Hipska — CHANGES_REQUESTED** [review]

File: helm-chart/templates/deployment.yaml (hunk 10 above)
Comment (Hipska) [reply helm-chart/templates/deployment.yaml]: See example: https://demo.combodo.com/simple/webservices/status.php

**Comment by Hipska** [comment]
Hi @finaldzn can you give some feedback on my questions/suggestions?

**Comment by finaldzn** [comment]
@Hipska im taking care of it today. Sorry for the delay

File: helm-chart/templates/NOTES.txt (hunk 9 above)
Comment (finaldzn) [reply helm-chart/templates/NOTES.txt]: this is the notes showed after the installation of the chart, not really sure what more can we put there

**Comment by finaldzn** [comment]
I updated it, not really sure what else we can do regarding the "notes.txt"

File: helm-chart/templates/NOTES.txt (hunk 9 above)
Comment (Hipska) [reply helm-chart/templates/NOTES.txt]: Okay, I have no experience with HELM, so I didn't know that. In this case it looks good to me.

File: helm-chart/README.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
+## HELM Chart
+
+This is a helm chart used to easily deploy iTop to a K8S cluster using [HELM](https://helm.sh/).
+
+### Volumes 
+In order for iTop to support being upgraded and being restarted pvc are automacaly created.
```
Comment (Hipska) [inline helm-chart/README.md]: Small typo:

```suggestion
### Volumes 

In order for iTop to support being upgraded and being restarted, pvc are automatically created.
```

**Review by Hipska — CHANGES_REQUESTED** [review]
Great improvements! There are still some open questions. Please give your feedback.

File: helm-chart/README.md (hunk 6 above)
Comment (finaldzn) [reply helm-chart/README.md]: Yeah not very clear, What i meant was we need to allow the itop service (so the php app ) to access to the files on the system, i updated it

Comment (Hipska) [reply helm-chart/README.md]: Okay thanks!

**Review by Hipska — APPROVED** [review]
Great, all seems clear now..

**Comment by finaldzn** [comment]
Great, what i can suggest is to host the chart in a chart museum so it can be as easy to install it as is the docker image

**Comment by Hipska** [comment]
Where do you suggest to add this?

**Comment by finaldzn** [comment]
Either using something like this : https://github.com/helm/chart-releaser-action#example-workflow

Or packaging and then pushing the chart to a public repository, but im not sure if docker hub is compatible (a helm chart implements OCI so its supposed to be compatible with docker hub but nerver tried it  )

**Pull request merged**